### ISSUE TITLE
World Hopping Bug

### ISSUE THREAD

**Issue by JM9877** [runelite/runelite#8133]
Please check if your issue is not a duplicate by [searching existing issues](https://github.com/runelite/runelite/search?type=Issues)

**Describe the bug**
So, I had a friend who had their private chat turned on but was playing runescape. They were listed as offline, but when i right clicked their name, I had the hop-to option available, so i selected it and it brought me to their world. Which shouldn't be possible if they had their private chat off, it should treat it like they are logged off.
**To Reproduce**

**Expected behavior**

**Screenshots**

**Environment (please complete the following information):**
 - OS: [e.g. Windows, Ubuntu, macOS]
 - RuneLite version: [e.g 1.4.6]
 - Launcher version: [e.g 1.5.2]

**Additional context**
Add any other context about the problem here, e.g. logs. Your client logs can usually be found in
`C:\Users\<your_user_name>\.runelite\logs` on Windows and `~/.runelite/logs` on Linux and macOS.

**Comment by chivesrs** [comment]
`had a friend who had their private chat turned on but was playing runescape`

Do you mean they had it turned OFF?

This looks like you hit an edge case when your friend had private off, but they were in a clan chat with you. There looks to be a bug where if the friend has private off and then leaves the clan chat, the hop-top option still shows when it should not.

**Comment by JM9877** [comment]
Yes, they had it off. We are in a clan chat together. They didnt leave the
clan chat though. Unless turning private off counts as leaving cc

**Comment by chivesrs** [comment]
Turning private off does not count as leaving the CC. I believe the behavior changes we want to make (as we discussed this in the Discord) are:

1. If the other player has private off, but is in the clan chat with you, don't show `hop-to` on the friend's list, only in the clan chat
2. If the other player has private off, and leaves the clan chat, clear the cached value of the world they are on

Doing 1. essentially does the same functionality as 2., but it's good to clean up the code such that 1. and 2. can function independently of each other (eg if we only did 2., then the bug would be fixed)

A third stretch goal:
3. If the other player is in your clan chat, but on another game (RS3, RS3 lobby), don't show `hop-to`. This is outside the immediate bug, but would be worth doing.
